src/metabase/automagic_dashboards/core.clj

```diff
@@ -632,16 +654,30 @@
 (def ^:private ^{:arglists '([x])} endocde-base64-json
   (comp codec/base64-encode codecs/str->bytes json/encode))
 
+(def ^:private ^{:arglists '([card-or-question])} nested-query?
+  (comp (every-pred string? #(str/starts-with? % "card__")) :source_table :query :dataset_query))
```

Comment:
You should use `metabase.query-processor.util/get-in-normalized` to extract things like this from the query map because it's supposed to be case-insensitive so it could come in as `SOURCE_QUERY` or something.